pkg/report/pdf.go

```diff
@@ -0,0 +1,477 @@
+	"github.com/johnfercher/maroto/pkg/consts"
+	"github.com/johnfercher/maroto/pkg/pdf"
+	"github.com/johnfercher/maroto/pkg/props"
+	"github.com/rs/zerolog/log"
+)
+
+const (
+	defaultTextSize = 8
+	pgMarginLeft    = 10
+	pgMarginTop     = 15
+	pgMarginRight   = 10
+	rowXSmall       = 3
+	rowSmall        = 4
+	rowMedium       = 5
+	rowLarge        = 8
+	rowXLarge       = 15
+	colOne          = 1
+	colTwo          = 2
+	colThree        = 3
+	colFour         = 4
+	colFive         = 5
+	colSix          = 6
+	colNine         = 9
+	colTen          = 10
+	colFullPage     = 12
+)
+
+var (
+	grayColor = getGrayColor()
+	//go:embed assets/vuln
+	vulnImageBase64 string
+)
+
+func createQueriesTable(m pdf.Maroto, queries []model.VulnerableQuery, basePath string) error {
+	for i := range queries {
+		m.SetBackgroundColor(color.NewWhite())
+		queryName := queries[i].QueryName
+		resultsCount := fmt.Sprint(len(queries[i].Files))
+		severity := string(queries[i].Severity)
+		platform := queries[i].Platform
+		category := queries[i].Category
+
+		var err error
+		m.Row(rowLarge, func() {
+			m.Col(colOne, func() {
+				err = m.Base64Image(vulnImageBase64, consts.Png, props.Rect{
+					Center:  false,
+					Percent: 50,
+					Left:    2,
+				})
+			})
+			m.Col(colNine, func() {
+				m.Text(queryName, props.Text{
+					Size:        11,
+					Style:       consts.Bold,
+					Align:       consts.Left,
+					Extrapolate: false,
+				})
+			})
+			m.Col(colOne, func() {
+				m.Text("Results", props.Text{
+					Size:        8,
+					Style:       consts.Bold,
+					Align:       consts.Right,
+					Extrapolate: false,
+				})
+			})
+			m.Col(colOne, func() {
+				m.Text(resultsCount, props.Text{
+					Size:        8,
+					Style:       consts.Bold,
+					Align:       consts.Right,
+					Extrapolate: false,
+				})
+			})
+		})
+		if err != nil {
+			return err
+		}
+		m.Row(colFour, func() {
+			m.Col(colTwo, func() {
+				m.Text("Severity", props.Text{
+					Size:        10,
+					Align:       consts.Left,
+					Extrapolate: false,
+				})
+			})
+			m.Col(colTwo, func() {
+				m.Text(severity, props.Text{
+					Size:        10,
+					Align:       consts.Left,
+					Extrapolate: false,
+				})
+			})
+		})
+		m.Row(colThree, func() {
+			m.Col(colTwo, func() {
+				m.Text("Platform", props.Text{
+					Size:        defaultTextSize,
+					Align:       consts.Left,
+					Extrapolate: false,
+				})
+			})
+			m.Col(colTwo, func() {
+				m.Text(platform, props.Text{
+					Size:        defaultTextSize,
+					Align:       consts.Left,
+					Extrapolate: false,
+				})
+			})
+		})
+		m.Row(colFive, func() {
+			m.Col(colTwo, func() {
+				m.Text("Category", props.Text{
+					Size:        defaultTextSize,
+					Align:       consts.Left,
+					Extrapolate: false,
+				})
+			})
+			m.Col(colTwo, func() {
+				m.Text(category, props.Text{
+					Size:        defaultTextSize,
+					Align:       consts.Left,
+					Extrapolate: false,
+				})
+			})
+		})
```

Comment:
I think we could extract this as function to be something like `createInfoComposition` and replace "Severity", "Platform" and Category row. I would suggest the following snippet to be the func:
```go
func createInfoComposition(m pdf.Maroto, label, info string, rowSize int, config props.Text) {
  m.Row(rowSize, func() {
    m.Col(colTwo, func() {
      m.Text(label, config)
    })
    m.Col(colTwo, func() {
       m.Text(info, config)
    })
  })
}
```
And replace as following:
```go
defaultTextConfig := props.Text{
  Size:        defaultTextSize,
  Align:       consts.Left,
  Extrapolate: false,
}
createInfoComposition(m, "Severity", severity, colFour, props.Text{
  Size:        10,
  Align:       consts.Left,
  Extrapolate: false,
})
createInfoComposition(m, "Platform", platform, colThree, defaultTextConfig)
createInfoComposition(m, "Category", category, colFive, defaultTextConfig)
```

Also, maybe `defaultTextConfig` could be a constant used on multiple locations in this file...

Reply:
good idea